### ISSUE TITLE
Patch for _refreshPosition issue in IE8

### ISSUE THREAD

**Issue by tschettler** [fnagel/jquery-ui#209]
In _refreshPosition and in very specific instances, the call to the position() jQuery UI function will double the listWrap's current top and left inline style values. This causes IE to throw the "can't move focus to control..." error and will not display the dropdown. After spending ~10 hours debugging the issue, I'm not sure if the problem is within the selectmenu plugin or in the position ui component, but this fixes it:

```
        this.listWrap
        .removeAttr('style')
        .zIndex(this.element.zIndex() + 1)
        .position({
            // set options for position plugin
            of: o.positionOptions.of || this.newelement,
            my: o.positionOptions.my,
            at: o.positionOptions.at,
            offset: o.positionOptions.offset || _offset,
            collision: o.positionOptions.collision || o.style == "popup" ? 'fit' : 'flip'
        });
```

This happens to the listWrap element (class ui-selectmenu-menu) on the position() call when you call selectmenu() after the menu is initialized. If the style attribute is removed, the doubling does not occur (nor does the IE8 error). I'm calling this a patch, because it is a workaround for the real issue, which I cannot figure out. I can get it to happen 100% of the time in IE8 on a page I'm developing, but have been unable to create a jsFiddle for it.

_jQuery 1.7.1, jQuery UI 1.8.17, and the latest selectmenu build_

Travis

**Comment by fnagel** [comment]
IE and the position mechanisms in this version of Selectmenu are pretty messed up. I guess this is your magic code snippet?

``` javascript
.removeAttr('style')
```

Any way to reproduce this? Or should I just test and trust you? In which browsers did you test? Did you use VM? Which OS?

**Comment by tschettler** [comment]
Yeah, that's the magic. I believe this is the root cause of the issue I was having with issue #191 and the fix in merge request #200.

I'm pretty sure it's an edge case, but it may fix other IE8 focus issues, including the issue #200 was attempting to resolve. I'm initializing the selectmenu multiple times for two instances, hiding options from the second instance based on the selected option of the first instance, and calling selectmenu again to update the options with the new css. It is this last selectmenu call that is causing the focus issue bug in IE8 again, while it doesn't happen on the other calls due to #200 fix.

In my experience, the position plugin can sometimes be finicky, I've had positioning issues when a FOUC exists. I'm not sure in this case why it doesn't just replace the top and left styles in IE8, rather than doubling them.

I've tested Chrome, Firefox, IE7/8, Opera, and Safari, as well as several mobile browsers. No VM, just Windows 7.

**Comment by fnagel** [comment]
Sorry for the late answer but I moved to another city.

I must confess I just realized youre using jQuery 1.7.1, which has some known issues: https://github.com/fnagel/jquery-ui/issues?labels=Update-jQuery-version&state=open
I hopefully will find some time this weekend to test this. Perhaps your fix is the key to 1.7.x compatibility!

**Comment by tschettler** [comment]
Sounds good, let me know if I can assist in any way.

**Comment by fnagel** [comment]
Ive updated Selectmenu to jQuery 1.7.2 and jQuery UI 1.8.20. There are some issues left, like fixing typeahead (see #213, help is appreciated!).

A general (not jQuery version related) problem with positioning in older IE was fixed before updating. This problem was introduced with the appendTo option and therefore only affects newer version of Selectmenu.

Would you please test if your problem could be related to the above mentioned issue? 
If not I will not ask again and just include your patch.

**Comment by tschettler** [comment]
After just updating my script references to jQuery 1.7.2 and jQuery UI 1.8.20, the issue is no longer present. It must have been something with the position api. I've updated my selectmenu reference and currently have no further issues to report.

Thanks!

**Comment by fnagel** [comment]
Great to hear! So I will close this as the issue is solved for now. Please open a new issue if needed.

**Comment by vincentpazeller** [comment]
Hello, I am sorry to disappoint you, but I just encountered that problem (may be of some different flavour),  although I am using jQuery 1.7.2 and jQuery UI 1.8.20. Nevertheless,  thank you to tschettler so much because I would probably still be banging my head against the wall. .removeAttr("style") worked like a charm for me (although I hate modifying libraries codes...). 

``` javascript
 $.get(Routing.generate("ICB_list_regions", {"country_id" : countryVal}), {}, function(regions){

            //clear regionField
            regionField = clearSelect(regionField);

            for(var i in regions){
                var region = regions[i];
                var option = document.createElement("OPTION");
                for(var j in region){
                    option.value = j;
                    option.text = region[j];
                    break;
                }
                regionField.options[regionField.options.length] = option;
            }
            //$(regionField).children('option');
            //toggleEnableField(regionField, true);

            $(regionField).selectmenu(); //refresh selectmenu widget
            onSuccess(validator, context);

        }, "json");

```

Basically what I am doing is filling a region selectmenu through ajax when the user selects a country (which is another selectmenu). No problem with other selects. The problem only occurs with the region "sub selectmenu" and specifically with IE8 (I did not test lower versions). the problem arises as soon as we try to open the region selectmenu after it has been refreshed with ajax.

I hope this will help you. Anyways thank you for your work!

**Comment by fnagel** [comment]
Thanks for your feedback. I think I will just include this, nevertheless Im confused :-/
One last question: did you used the latest version?

**Comment by vincentpazeller** [comment]
I think I downloaded 2 days ago (might be 4 days because I have a bad memory ^^) so I would say yes... Sorry for this bad new s:s

**Comment by fnagel** [comment]
Its ok, I will just include this and hopefully never hear of problems like this again :-)

This version of Selectmenu has limited lifetime anyway as the new, official jQuery UI 1.9 Selectmenu widget is ready for use and will hopefully released in a few month. See #140 for more infos.

**Comment by vincentpazeller** [comment]
Ok, good to hear. Dealing with versions will be easier :)

**Comment by tschettler** [comment]
I think I spoke too soon, I updated to the latest version without my fix and a user reported the issue to me. Looks like I'll be putting it back in again.

**Comment by fnagel** [comment]
Thanks for the feedback Travis. As already mentioned I will include your
fix in my next dev cycle!

**Comment by tschettler** [comment]
Yep, sounds good, thanks again!

**Comment by fnagel** [comment]
Fix in latest commit. Thanks for your contribution. 

Please reopen if needed!
